std-bits/table/src/main/java/org/enso/table/data/table/Column.java

```diff
@@ -158,27 +157,7 @@ public static Column fromItemsNoDateConversion(
    */
   public static Column fromRepeatedItem(
       String name, Value item, int repeat, ProblemAggregator problemAggregator) {
-    if (repeat < 0) {
-      throw new IllegalArgumentException("Repeat count must be non-negative.");
-    }
-
-    Object converted = Polyglot_Utils.convertPolyglotValue(item);
-
-    if (converted == null) {
-      var builder = Builder.getForType(AnyObjectType.INSTANCE, repeat, problemAggregator);
-      builder.appendNulls(repeat);
-      return new Column(name, builder.seal());
-    }
-
-    StorageType storageType = StorageType.forBoxedItem(converted);
-    Builder builder = Builder.getForType(storageType, repeat, problemAggregator);
-    Context context = Context.getCurrent();
-    for (int i = 0; i < repeat; i++) {
-      builder.append(converted);
-      context.safepoint();
-    }
-
-    return new Column(name, builder.seal());
+    return new Column(name, Storage.fromRepeatedItem(item, repeat, problemAggregator));
```

Comment:
@jdunkerley I've allowed myself to overwrite some of your cleanups from the other PR in this particular method with a fully custom `fromRepeatedItem`.

It also relies on `Builder.getForType` for the generic case but also has 'specializations' that create a more compact storage directly (`NullStorage` and `LongConstantStorage` that are constructed in O(1) instead of O(N)).

I think we do want this 'optimization'. In the future we can even add more `ConstantStorage` kinds. 

The major question is if `Storage` class is the right place for this method. I imagine once we start adding more backends it may not necessarily be. But for now I think it is relatively good place. So I'd keep it here and move it when it makes sense.

Does that seem OK?

Reply:
Yes fine to do. Will rebase and cope :)